contracts/src/LightClient.sol

```diff
@@ -373,6 +402,51 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
         }
     }
 
+    /// @notice updates the stateHistoryCommitments array each time a new
+    /// finalized state is added to the LightClient contract.
+    /// Ensures that the array contains only the most recent data and does not
+    /// exceed the maxStateHistoryDuration duration in seconds.
```

Comment:
I think we don't do this exactly since we at most remove one element.

Reply:
addressed here https://github.com/EspressoSystems/espresso-sequencer/commit/23a7bdc537f956583d6e57b5ce64c03623890899